bundles/http/Citrix/SAPHana/script.js

```diff
@@ -0,0 +1,921 @@
+        "client": client,
+        "receiverCostCenter": workAgreement.CostCenter,
+        "purchaseOrderItem": actionParameters.PurchaseOrderItem,
+        "timeSheetTaskType": actionParameters.TimeSheetTaskType,
+        "timeSheetTaskLevel": actionParameters.TimeSheetTaskLevel,
+        "timeSheetTaskComponent": actionParameters.TimeSheetTaskComponent,
+        "timeSheetNote": actionParameters.TimeSheetNote,
+        "recordedHours": recordedTime.toString(),
+        "hoursUnitOfMeasure": actionParameters.HoursUnitOfMeasure,
+        "personWorkAgreementExternalID": workAgreement.PersonWorkAgreementExternalID,
+        "companyCode": workAgreement.CompanyCode,
+        "personWorkAgreement": workAgreement.PersonWorkAgreement,
+        "date": date,
+        "timeSheetStatus": actionParameters.TimeSheetStatus,
+        "timeSheetOperation": actionParameters.TimeSheetOperation,
+        "timeEntryId": actionParameters.TimeEntryId,
+        "startedAt": null,
+        "endAt": '',
+    }
+
+    const createdEntry = await createTimeEntryApiRequest(createTimeEntryApiRequestObject)
+
+    const timeEntry = await getCreatedTimeEntryApiRequest(client, createdEntry.d.PersonWorkAgreementExternalID,
+        createdEntry.d.CompanyCode, createdEntry.d.TimeSheetRecord)
+
+    const timeSheetDateTimeStamp = timeEntry.TimeSheetDate.replace(/[^.\d]/g, '');
+
+    const storeObject = {
+        "dataStore": dataStore,
+        "timeEntryId": null,
+        "timeSheetRecord": timeEntry.TimeSheetRecord,
+        "timeSheetDateTimeStamp": timeSheetDateTimeStamp,
+        "companyCode": timeEntry.CompanyCode,
+        "personWorkAgreement": timeEntry.PersonWorkAgreement,
+        "personWorkAgreementExternalID": timeEntry.PersonWorkAgreementExternalID,
+        "controllingArea": timeEntry.TimeSheetDataFields.ControllingArea,
+        "hoursUnitOfMeasure": timeEntry.TimeSheetDataFields.HoursUnitOfMeasure,
+        "purchaseOrderItem": timeEntry.TimeSheetDataFields.PurchaseOrderItem,
+        "receiverCostCenter": timeEntry.TimeSheetDataFields.ReceiverCostCenter, //
+        "recordedHours": timeEntry.TimeSheetDataFields.RecordedHours,
+        "recordedQuantity": timeEntry.TimeSheetDataFields.RecordedQuantity,
+        "timeSheetOvertimeCategory": timeEntry.TimeSheetDataFields.TimeSheetOvertimeCategory,
+        "timeSheetTaskComponent": timeEntry.TimeSheetDataFields.TimeSheetTaskComponent,
+        "timeSheetTaskLevel": timeEntry.TimeSheetDataFields.TimeSheetTaskLevel,
+        "timeSheetTaskType": timeEntry.TimeSheetDataFields.TimeSheetTaskType,
+        "timeSheetNote": timeEntry.TimeSheetDataFields.TimeSheetNote,
+        "timeSheetOperation": timeEntry.TimeSheetOperation,
+        "timeSheetStatus": timeEntry.TimeSheetStatus,
+        "user_email": actionParameters.UserEmail,
+        "startedAt": actionParameters.StartedAt,
+        "timeClockManagement": isTimeClockManagement,
+        "endAt": '',
+    }
+
+    await storeEntryInDatabase(storeObject)
```

Comment:
```suggestion
   storeEntryInDatabase(storeObject)
```
await not needed here. There is no code that needs a result from this function

Reply:
await removed